File: backend/core/src/modules/beta-feedback/beta-feedback.controller.ts
import { Body, Controller, Delete, Get, Param, Post, Query, Req, UseGuards } from '@nestjs/common';
import { JwtAuthGuard } from '../auth/jwt-auth.guard';
import { BetaFeedbackService } from './beta-feedback.service';

@UseGuards(JwtAuthGuard)
@Controller('beta-feedback')
export class BetaFeedbackController {
  constructor(private readonly betaFeedbackService: BetaFeedbackService) {}

  @Get()
  findAll(@Req() req: any, @Query('sessionId') sessionId?: string) {
    if (!sessionId) return [];
    return this.betaFeedbackService.findAllForUser(req.user?.userId, sessionId);
  }

  @Get(':id')
  findOne(@Req() req: any, @Param('id') id: string) {
    return this.betaFeedbackService.findOneForUser(req.user?.userId, id);
  }

  @Post()
  create(@Req() req: any, @Body() dto: any) {
    return this.betaFeedbackService.createForUser(req.user?.userId, dto);
  }

  @Delete(':id')
  async remove(@Req() req: any, @Param('id') id: string) {
    await this.betaFeedbackService.removeForUser(req.user?.userId, id);
    return { success: true };
  }
}

File: backend/core/src/main/java/store/pjcloud/cowrite/core/controller/BetaFeedbackController.java
package store.pjcloud.cowrite.core.controller;

import java.util.List;
import java.util.Map;
import java.util.UUID;
import org.springframework.web.bind.annotation.DeleteMapping;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;
import store.pjcloud.cowrite.core.security.SecurityUtils;
import store.pjcloud.cowrite.core.service.BetaFeedbackService;

@RestController
@RequestMapping("/beta-feedback")
public class BetaFeedbackController {
    private final BetaFeedbackService betaFeedbackService;

    public BetaFeedbackController(BetaFeedbackService betaFeedbackService) {
        this.betaFeedbackService = betaFeedbackService;
    }

    @GetMapping
    public List<Map<String, Object>> findAll(@RequestParam(value = "sessionId", required = false) UUID sessionId) {
        UUID userId = SecurityUtils.requireUserId();
        if (sessionId == null) return List.of();
        return betaFeedbackService.findAllForUser(userId, sessionId);
    }

    @GetMapping("/{id}")
    public Map<String, Object> findOne(@PathVariable("id") UUID id) {
        UUID userId = SecurityUtils.requireUserId();
        return betaFeedbackService.findOneForUser(userId, id);
    }

    @PostMapping
    public Map<String, Object> create(@RequestBody Map<String, Object> body) {
        UUID userId = SecurityUtils.requireUserId();
        return betaFeedbackService.createForUser(userId, body);
    }

    @DeleteMapping("/{id}")
    public Map<String, Object> remove(@PathVariable("id") UUID id) {
        UUID userId = SecurityUtils.requireUserId();
        betaFeedbackService.removeForUser(userId, id);
        return Map.of("success", true);
    }
}
